### PR TITLE
trigger monitoring : orbit vs bx

### PR DESCRIPTION
**Pull request by mtosi**
as PR #15813

### CODE REVIEW

**Review by dmitrijus — APPROVED** [review]

**Comment by dmitrijus** [comment]
+1

**Comment by fwyzard** [comment]
hold

**Comment by fwyzard** [comment]
assign hlt

**Comment by fwyzard** [comment]
@mtosi sorry for being picky, but could you please clean up the commit history ?
  - remove all changes to TriggerBxMonitor.cc and triggerBxMonitor.py
  - introduce the new files with a single commit

(I would say that the easiest approach is to close this PR, and make a new one where you simply add the new files)

**Comment by davidlange6** [comment]
assign hlt

**Comment by Martin-Grunewald** [comment]
@fwyzard
You have pout this PR on hold - what is the remaining hold-up?

**Comment by mtosi** [comment]
I think this PR is fine, now ...isn't it @fwyzard ?
thanks

**Comment by fwyzard** [comment]
please test

**Comment by fwyzard** [comment]
+1

File: DQM/HLTEvF/plugins/TriggerBxVsOrbitMonitor.cc [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    // book 2D histogram to monitor all TCDS trigger types in a single plot
+    booker.setCurrentFolder( m_dqm_path + "/orbitVsBX" );
+    m_orbit_bx_all = booker.book2D("OrbitVsBX", "Event orbits vs. bunch crossing", nBX, float(m_minBX)-0.5, float(m_maxBX)+0.5, s_orbit_range+1, -0.5, float(s_orbit_range)+0.5)->getTH2F();
+    m_orbit_bx_all->GetXaxis()->SetTitle("BX");
+    m_orbit_bx_all->GetYaxis()->SetTitle("orbit");
+    m_orbit_bx_all->SetCanExtend(TH1::kAllAxes);
+    
+    for (unsigned int i = 0; i < nLS; ++i) {
+      std::string iname = boost::lexical_cast<std::string>(i);
+      m_orbit_bx_all_byLS.at(i) = booker.book2D("OrbitVsBX_LS"+iname, "OrbitVsBX_LS"+iname, nBX, float(m_minBX)-0.5, float(m_maxBX)+0.5, s_orbit_range+1, -0.5, float(s_orbit_range)+0.5)->getTH2F();
```
Comment (davidlange6) [inline DQM/HLTEvF/plugins/TriggerBxVsOrbitMonitor.cc]: so do I compute correctly this is 3 ls * 18 bx * 260k orbits or 60MB?

**Comment by mtosi** [comment]
ciao
yeah :(
indeed it needs to have a limited number of BXs and LSs as well
in addition, such a plot will not go in standard production

if you have any suggestion, it would be really appreciated ;)
thanks
  mia

**Comment by fwyzard** [comment]
Hi David,
as a side note, the idea is to *not* run this in production, but only in
user-defined jobs, with the configuration on BX etc adapted to the
individual use cases.

Ciao,
.Andrea

**Comment by davidlange6** [comment]
+1

**Pull request merged**